runtime/caml/memory.h

```diff
@@ -231,8 +231,18 @@ enum caml_alloc_small_flags {
 #define Alloc_small(result, wosize, tag, GC) \
   Alloc_small_with_reserved(result, wosize, tag, GC, (uintnat)0)
 
-CAMLextern caml_local_arenas* caml_get_local_arenas(caml_domain_state*);
-CAMLextern void caml_set_local_arenas(caml_domain_state*, caml_local_arenas* s);
+// Retrieve the local arenas for the given stack.
+// If the stack is the current stack, the copy of [local_sp] at the
+// root of [Caml_state] is saved in the current [stack_info]
+// structure, as it may have been updated by OCaml code.
+CAMLextern caml_local_arenas* caml_get_local_arenas_and_save_local_sp(struct stack_info*);
```

Comment:
I don't like the name of this function!

  - it's long and describes the implementation rather than what it's for
  - it sounds like the getter corresponding to `caml_set_local_arenas` but it's not (that one only sets the current stack's arenas, but this one reads the arenas of the given stack)

Reply:
How about `caml_refresh_locals`? Also changed `set` to `use`.

Reply:
Names seem fine now.